### PR TITLE
Replacing <Avatar/> component by native one

### PR DESCRIPTION
**Pull request by aleksfront**
* Removing Material UI avatar component for sake of #4138 
* Using raw avatar instead of `<HotbarIcon/>` in most of the places
* Moving `Avatar`, `HotbarIcon` and `HotbarEntityIcon` styles into css modules

https://user-images.githubusercontent.com/9607060/143874359-f0407e0e-3932-4e4c-a5c5-3e5cec34cda2.mov

### CODE REVIEW

File: src/renderer/components/+catalog/catalog-entity-details.tsx [hunk 1]
```diff
@@ -27,10 +27,10 @@ import type { CatalogCategory, CatalogEntity } from "../../../common/catalog";
 import { Icon } from "../icon";
 import { CatalogEntityDrawerMenu } from "./catalog-entity-drawer-menu";
 import { CatalogEntityDetailRegistry } from "../../../extensions/registries";
-import { HotbarIcon } from "../hotbar/hotbar-icon";
 import type { CatalogEntityItem } from "./catalog-entity-item";
 import { isDevelopment } from "../../../common/vars";
 import { cssNames } from "../../utils";
+import { Avatar } from "../avatar/avatar";
```
Comment (Nokel81) [inline src/renderer/components/+catalog/catalog-entity-details.tsx]: Shouldn't `components/avatar/index.ts` exist?

File: src/renderer/components/avatar/avatar.module.css [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+}
+
+.circle {
+  border-radius: 50%;
+}
+
+.rounded {
+  border-radius: var(--border-radius);
+}
```
Comment (Nokel81) [inline src/renderer/components/avatar/avatar.module.css]: I think that the board-radius is not large enough. I liked the previous border-radius for "rounded" square more.

File: src/renderer/components/hotbar/hotbar-icon.tsx [hunk 3]
```diff
@@ -75,33 +70,25 @@ export const HotbarIcon = observer(({ menuItems = [], size = 40, tooltip, ...pro
...
-          className={cssNames(active ? "active" : "default", { interactive: !!onClick })}
-          width={size}
-          height={size}
-          src={src}
-        >
-          {material && <Icon className="materialIcon" material={material} />}
-        </Avatar>
-        {children}
-      </div>
+        {material && <Icon material={material} />}
```
Comment (Nokel81) [inline src/renderer/components/hotbar/hotbar-icon.tsx:86]: What if `material` is an svg?

Comment (aleksfront) [reply src/renderer/components/hotbar/hotbar-icon.tsx:86]: Currently entities support only string icon notation and icons being used for General catalog entities, like here https://github.com/lensapp/lens/blob/master/src/common/catalog-entities/general.ts#L60

File: src/renderer/components/avatar/avatar.module.css (hunk 2 above)
Comment (aleksfront) [reply src/renderer/components/avatar/avatar.module.css]: Adjusted `border-radius` with more rounded value. I agreed it looks better now:
rounded avatars

File: src/renderer/components/hotbar/hotbar-icon.tsx (hunk 3 above)
Comment (Nokel81) [reply src/renderer/components/hotbar/hotbar-icon.tsx:86]: What about KubernetesCluster entities https://github.com/lensapp/lens/blob/master/src/common/catalog-entities/kubernetes-cluster.ts#L49-L54?

Comment (aleksfront) [reply src/renderer/components/hotbar/hotbar-icon.tsx:86]: This line `src?: string` relates to image URL and gets passed to `<Avatar src={src}/>` directly.

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**